src/main/java/com/google/sps/data/Referendum.java

```diff
@@ -69,9 +77,20 @@ public static Referendum fromJSONObject(JSONObject obj) throws JSONException {
       referendumDescription = "";
     }
 
+    if (obj.has(SOURCE_JSON_KEYWORD)) {
+      StringBuilder strBuf = new StringBuilder();
+      for (Object sourceObject : obj.getJSONArray(SOURCE_JSON_KEYWORD)) {
```

Comment:
This is probably fine.  Though I personally love streams/maps.

I think I would consider using a StringJoiner instead, so that you don't need to have the code below for stripping off the last comma.

https://docs.oracle.com/javase/8/docs/api/java/util/StringJoiner.html

Alternatively, you could also consider storing this as a list and construct the comma separated list on client side.